power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/buffer.hpp

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_BUFFER_HPP
+#define POWER_GRID_MODEL_CPP_BUFFER_HPP
+
+#include "buffer.h"
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+namespace power_grid_model_cpp {
+class Buffer {
+  public:
+    Buffer(PGM_MetaComponent const* component, Idx size)
+        : component_{component}, size_{size}, buffer_{PGM_create_buffer(handle_.get(), component, size)} {};
+
+    ~Buffer() = default;
+
+    static void set_nan(Handle const& handle, PGM_MetaComponent const* component, void* buffer_ptr, Idx buffer_offset,
+                        Idx size) {
+        PGM_buffer_set_nan(handle.get(), component, buffer_ptr, buffer_offset, size);
+        handle.check_error();
+    }
+    void set_nan(Idx buffer_offset) {
+        PGM_buffer_set_nan(handle_.get(), component_, buffer_.get(), buffer_offset, size_);
+        handle_.check_error();
+    }
```

Comment:
here and elsewhere: for readability, brevity and maintainability: please use one from the other.

Either:

```suggestion
    static void set_nan(Handle const& handle, PGM_MetaComponent const* component, void* buffer_ptr, Idx buffer_offset,
                        Idx size) {
        PGM_buffer_set_nan(handle.get(), component, buffer_ptr, buffer_offset, size);
        handle.check_error();
    }
    void set_nan(Idx buffer_offset) {
        set_nan(handle_.get(), *this, buffer_.get(), buffer_offset, size_);
    }
```

or

```suggestion
    static void set_nan(Handle const& handle, PGM_MetaComponent const* component, void* buffer_ptr, Idx buffer_offset,
                        Idx size) {
        component.set_nan(buffer_ptr, buffer_offset, size);
    }
    void set_nan(Idx buffer_offset) {
        PGM_buffer_set_nan(handle_.get(), component_, buffer_.get(), buffer_offset, size_);
        handle_.check_error();
    }
```

Reply:
You are correct. Big oversight from my side. I will stick with the first version, since I find it more natural.